basket/base/tests/test_rq_utils.py

```diff
@@ -0,0 +1,239 @@
+        """
+        with patch("basket.base.rq.random") as mock_random:
+            mock_random.randrange.side_effect = [2 * 2**n for n in range(settings.RQ_MAX_RETRIES)]
+            self.assertEqual(rq_exponential_backoff(), [60, 60, 60, 60, 60, 64, 128, 256, 512, 1024])
+
+    @override_settings(RQ_MAX_RETRIES=10, DEBUG=True)
+    def test_rq_exponential_backoff_with_debug(self):
+        """
+        Test that the exponential backoff function returns shorter retries during DEBUG mode.
+        """
+        self.assertEqual(rq_exponential_backoff(), [5, 5, 5, 5, 5, 5, 5, 5, 5, 5])
+
+    def test_get_redis_connection(self):
+        """
+        Test that the get_redis_connection function returns a Redis connection with params we expect.
+        """
+        # Test passing a URL explicitly.
+        connection = get_redis_connection("redis://redis-host:6379/9", force=True)
+        self.assertDictEqual(connection.connection_pool.connection_kwargs, {"host": "redis-host", "port": 6379, "db": 9})
+
+        # Test with no URL argument, but with RQ_URL in the environment.
+        # Note: Set this back to the "default" for tests that follow.
+        with patch.dict(os.environ, {"RQ_URL": "redis://redis:6379/2"}):
+            connection = get_redis_connection(force=True)
+            self.assertDictEqual(connection.connection_pool.connection_kwargs, {"host": "redis", "port": 6379, "db": 2})
+
+    def test_get_queue(self):
+        """
+        Test that the get_queue function returns a RQ queue with params we expect.
+        """
+        queue = get_queue()
+
+        assert queue.name == "default"
+        assert queue._is_async is False  # Only during testing.
+        assert queue.connection == get_redis_connection()
+        assert queue.serializer == JSONSerializer
+
+    def test_get_worker(self):
+        """
+        Test that the get_worker function returns a RQ worker with params we expect.
+        """
+        worker = get_worker()
+        assert worker.queues == [get_queue()]
+        assert worker.disable_default_exception_handler is True
+        assert worker._exc_handlers == [store_task_exception_handler]
+        assert worker.serializer == JSONSerializer
+
+    @override_settings(STORE_TASK_FAILURES=True)
+    @override_settings(RQ_EXCEPTION_HANDLERS=["basket.base.rq.store_task_exception_handler"])
+    @patch("basket.base.rq.statsd")
+    def test_on_failure(self, mock_statsd):
+        """
+        Test that the on_failure function creates a FailedTask object and sends
+        statsd metrics.
+        """
+
+        assert FailedTask.objects.count() == 0
+
+        args = ["arg1"]
+        kwargs = {"arg2": "foo"}
+        job = failing_job.delay(*args, **kwargs)
+
+        worker = get_worker()
+        worker.work(burst=True)  # Burst = worker will quit after all jobs consumed.
+
+        assert job.is_failed
+
+        # TODO: Determine why the `store_task_exception_handler` is not called.
```

Comment:
Is this a blocker or OK as a known-unknown?

Reply:
I added more tests so there's 100% coverage, it's just broken into separate tests. But I'd like to solve this riddle at some point. So I don't consider it a blocker, just a hard thing to test.